### PR TITLE
test: e2e scenarios

### PR DESCRIPTION
**Pull request by 0xnigir1**
# 🤖 Linear

Closes GIT-256

## Description
We add e2e scenarios for:
- Search projects DB function
- Round creation with Pending Roles
- Unsupported strategies event handling
- Application registration to a Round
- Donations to a Round and Application
- Direct donations to a Round
- Attestations on different chains or without linked donation

## Summary by CodeRabbit

- **New Features**
  - Introduced a robust project search that delivers case-insensitive, relevance-based results.
  - Streamlined round creation and application processing for a smoother project participation experience.
  - Enhanced event filtering based on chain ID for improved data handling.
  - Added validation for metadata retrieval to prevent processing of invalid content identifiers.
  - Improved support for multi-chain configurations in processing services.

- **Bug Fixes**
  - Enhanced metadata validation to gracefully handle invalid content identifiers.
  - Refined the processing of allocation events to ensure more accurate donation and payout records.

- **Tests**
  - Added comprehensive test suites for allocation events, round applications, project searches, and attestation scenarios to validate system behavior and edge cases.
  - Improved test coverage for metadata validation and multi-chain configurations.

### CODE REVIEW

**Comment by 0xnigir1** [comment]
@coderabbitai full review

File: tests/e2e/src/mocks/envio-graphql-server.mock.ts [hunk 1]
```diff
@@ -78,17 +78,31 @@ export class MockEnvioIndexer {
...
             if (query.includes("getEventsAfterBlockNumberAndLogIndex")) {
-                const events = [...this.events];
-                this.events = [];
+                // remove filtered events from the original events array
+                this.events = this.events.filter((event) => !events.includes(event));
                 res.json({ data: { raw_events: events } });
                 return;
             }
 
             if (query.includes("getEvents")) {
```
Comment (jahabeebs) [inline tests/e2e/src/mocks/envio-graphql-server.mock.ts]: You can just add this if condition to the above if statement since they have the same content

Comment (0xnigir1) [reply tests/e2e/src/mocks/envio-graphql-server.mock.ts]: [f418395](https://github.com/defi-wonderland/grants-stack-indexer-v2/pull/84/commits/f418395d890d6a7f43a845680ebddc84ced1591a)

File: tests/e2e/src/mocks/envio-graphql-server.mock.ts [hunk 2]
```diff
@@ -104,6 +92,32 @@ export class MockEnvioIndexer {
...
+
+            // Filter events by chainId if specified
+            const events = chainId
+                ? [...this.events].filter((event) => event.chainId === chainId)
+                : [...this.events];
+
+            // remove filtered events from the original events array
+            this.events = this.events.filter((event) => !events.includes(event));
+
+            if (query.includes("getEventsAfterBlockNumberAndLogIndex")) {
```
Comment (jahabeebs) [inline tests/e2e/src/mocks/envio-graphql-server.mock.ts]:
```suggestion
            if (query.includes("getEventsAfterBlockNumberAndLogIndex")  || query.includes("getEvents")) {
```

File: tests/e2e/src/mocks/envio-graphql-server.mock.ts [hunk 3]
```diff
@@ -104,6 +92,32 @@ export class MockEnvioIndexer {
...
+                res.json({ data: { raw_events: events } });
+                return;
+            }
+
+            if (query.includes("getEvents")) {
+                res.json({ data: { raw_events: events } });
+                return;
+            }
```
Comment (jahabeebs) [inline tests/e2e/src/mocks/envio-graphql-server.mock.ts]:
```suggestion
```

**Review by jahabeebs — COMMENTED** [review]
sorry if I wasn't clear--I meant this

File: tests/e2e/src/mocks/envio-graphql-server.mock.ts (hunk 2 above)
Comment (0xnigir1) [reply tests/e2e/src/mocks/envio-graphql-server.mock.ts]: hehe fixed again [5e8e3ca](https://github.com/defi-wonderland/grants-stack-indexer-v2/pull/84/commits/5e8e3ca75b852ccbeb68cb55b6d90c44029a4502)

**Review by jahabeebs — DISMISSED** [review]

**Review by jahabeebs — APPROVED** [review]

**Review by 0xkenj1 — APPROVED** [review]
